firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/BreathingCircuit/ControlLoop.h

```diff
@@ -19,9 +19,13 @@
 
 namespace Pufferfish::Driver::BreathingCircuit {
 
+static constexpr auto alarm_codes = Util::Containers::make_array<LogEventCode>(
+    LogEventCode::LogEventCode_sfm3019_air_disconnected,
+    LogEventCode::LogEventCode_sfm3019_o2_disconnected);
+
 class ControlLoop {
  public:
-  virtual void update(uint32_t current_time) = 0;
+  virtual void update(uint32_t current_time, Application::AlarmsManager &alarms_manager) = 0;
```

Comment:
Is there any way we can avoid making the `ControlLoop` classes be dependent on (or aware of) `AlarmsManager`? For example, perhaps we could just write `ControlLoop` sensor connected/disconnected `bool`s to a struct which is provided as an output parameter, and then have a `BreathingCircuit::SensorAlarmsService` which transforms the struct into `AlarmsManger` changes. Compare this proposed approach to how the `Controller` class writes things to the `actuator_setpoints` and `actuator_vars` output parameters. (In this proposed approach, we might rename our current `BreathingCircuit::AlarmsService` to `BreathingCircuit::AirwayAlarmsService` or something; later, we would also move the SpO2 and HR alarms out of the `BreathingCircuit` driver and into the `Nonin` driver).

Reasons for this:
- `ControlLoop`'s only responsibility should be to get readings from sensors and output actuation values, not to activate/deactivate alarms.
- `ControlLoop` is going to become much more complicated in the future, so let's keep it as simple as possible.
- Activation of other alarms has already been decomposed out to other classes, so let's continue separating alarm activation/deactivation into other classes.
- By reducing interdependencies, we can simplify unit tests - so that the unit tests for `ControlLoop` are totally independent of `AlarmsManager`

Currently `ParametersService` does depend on `LogManager` as well as `AlarmsManager`; if we can find a way to decouple them, we should.